doc/api/crypto.md

```diff
@@ -1671,6 +1671,110 @@ Use [`crypto.getHashes()`][] to obtain an array of names of the available
+  - `divisorLength`: {number} size of `q` in bits (DSA)
+  - `namedCurve`: {string} name of the curve to use (EC)
+  - `publicKeyEncoding`: {Object}
+    - `type`: {string} Must be one of `'pkcs1'` (RSA only) or `'spki'`
+    - `pem`: {boolean} Whether to encode the key as PEM or as DER
+  - `privateKeyEncoding`: {Object}
+    - `type`: {string} Must be one of `'pkcs1'` (RSA only), `'pkcs8'` or
+      `'sec1'` (EC only)
+    - `pem`: {boolean} Whether to encode the key as PEM or as DER
+    - `cipher`: {string} If specified, the private key will be encrypted with
+      the given `cipher` and `passphrase` using PKCS#5 v2.0 password based
+      encryption.
+    - `passphrase`: {string} The passphrase to use for encryption, see `cipher`.
+* `callback`: {Function}
+  - `err`: {Error}
+  - `publicKey`: {string|Buffer}
+  - `privateKey`: {string|Buffer}
+
+Generates a new asymmetric key pair of the given `type`. Only RSA, DSA and EC
+are currently supported.
+
+It is recommended to encode public keys as `'spki'` and private keys as
+`'pkcs8'` with encryption:
+
+```js
+const { generateKeyPair } = require('crypto');
+generateKeyPair('rsa', {
+  modulusLength: 4096,
+  publicKeyEncoding: {
+    type: 'spki',
+    pem: true
+  },
+  privateKeyEncoding: {
+    type: 'pkcs8',
+    pem: true,
+    cipher: 'aes-256-cbc',
+    passphrase: 'top secret'
+  }
+}, (err, publicKey, privateKey) => {
+  // Handle errors and use the generated key pair.
+});
+```
+
+On completion, `callback` will be called with `err` set to `undefined` and
+`publicKey` / `privateKey` representing the generated key pair. When PEM
+encoding was selected, the result will be a string, otherwise it will be a
+buffer containing the data encoded as DER. Note that Node.js itself does not
+accept DER, it is supported for interoperability with other libraries such as
+WebCrypto only.
+
+### crypto.generateKeyPairSync(type, options)
+<!-- YAML
+added: REPLACEME
+-->
+* `type`: {string} Must be `'rsa'`, `'dsa'` or `'ec'`.
+* `options`: {Object}
+  - `modulusLength`: {number} key size in bits (RSA, DSA)
+  - `publicExponent`: {number} public exponent (RSA)
+  - `divisorLength`: {number} size of `q` in bits (DSA)
+  - `namedCurve`: {string} name of the curve to use (EC)
+  - `publicKeyEncoding`: {Object}
+    - `type`: {string} Must be one of `'pkcs1'` (RSA only) or `'spki'`
+    - `pem`: {boolean} Whether to encode the key as PEM or as DER
+  - `privateKeyEncoding`: {Object}
+    - `type`: {string} Must be one of `'pkcs1'` (RSA only), `'pkcs8'` or
+      `'sec1'` (EC only)
+    - `pem`: {boolean} Whether to encode the key as PEM or as DER
+    - `cipher`: {string} If specified, the private key will be encrypted with
+      the given `cipher` and `passphrase` using PKCS#5 v2.0 password based
+      encryption.
+    - `passphrase`: {string} The passphrase to use for encryption, see `cipher`.
+* Returns: {Array} An array of the form `[publicKey, privateKey]`
```

Comment:
Can the `publicKey, privateKey` be a string or buffer too? If so, the type probably should be `{string[]|Buffer[]}` and the note should be added at the description end like in the previous description.

Reply:
@vsemozhetbyt Yes, they can, but the `pem` setting is per key type. Would that be `{(string|Buffer)[]}`?

Reply:
I am afraid our type parser does not support this variant. Maybe leave the `{Array}` then and add a note?